### ISSUE TITLE
Datastore_search doesn't allow "hybrid" searches

### ISSUE THREAD

**Issue by jqnatividad** [ckan/ckan#5975]
**CKAN version**
master

**Describe the bug**
The `datastore_search` q parameter can either do a table-wide FTS query OR a per-field query, but not both.  Per the [API doc](https://docs.ckan.org/en/2.9/maintaining/datastore.html#ckanext.datastore.logic.action.datastore_search):

> q (string or dictionary) – full text query. If it’s a string, it’ll search on all fields on each row. If it’s a dictionary as {“key1”: “a”, “key2”: “b”}, it’ll search on each specific field (optional)

**Expected behavior**
A table-wide FTS query combined with per-field query should be possible. Right now, the validator prevents the internal "_full_text" field from being added as a per-field query which should make this possible without changing the API interface.

**Comment by wardi** [comment]
Let's add a new parameter for full text search, how about:
- full_text (string) – full text query search on all fields on each row (optional)
- q (dictionary) – a dictionary as {“key1”: “a”, “key2”: “b”}, will match records on each specific field (optional). if a string is passed treat it like it was sent to the full_text parameter, for backwards compatibility only.

**Comment by jqnatividad** [comment]
That approach looks good to me.

It might be worthwhile to mention in the apidoc  that when both are specified, the search result set will be the UNION (logical OR) of both search parameters' result sets.

**Comment by wardi** [comment]
why not AND? That's how the column filters work, right?

**Comment by wardi** [comment]
it's also kind of strange UI to add a search and get more results back than before

**Comment by jqnatividad** [comment]
You're right, it should be the INTERSECTION (logical AND) of both result sets.

I just got confused by the current search highlighting behavior of datatables_view.  Right now, if you do both a table-wide and a column filter search, it will apply search highlighting for both even though it only effectively uses the column filter, giving the user the false impression that its using both filters.

Doing it as you proposed (AND), will make the search highlighting behavior reflect what's actually retrieved.

FYI, in an earlier version of the revamped datables view, I even had some javascript making col and table filters mutually exclusive and dynamically greying out the filters as required.  It unnecessarily complicated the code and the UX so I removed it.
